### PR TITLE
bg_haka_tomb OK

### PR DESCRIPTION
**Pull request by kyleburnette**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Haka_Tomb/z_bg_haka_tomb.c [hunk 1]
```diff
@@ -9,7 +9,16 @@ void BgHakaTomb_Destroy(Actor* thisx, GlobalContext* globalCtx);
...
 
-/*
+void func_80BD6624(BgHakaTomb* this);
+void func_80BD66AC(BgHakaTomb* this, GlobalContext* globalCtx);
+void func_80BD6754(BgHakaTomb* this);
+void func_80BD6768(BgHakaTomb* this, GlobalContext* globalCtx);
+void BgHakaTomb_SetupDoNothing(BgHakaTomb* this);
+void BgHakaTomb_DoNothing(BgHakaTomb* this, GlobalContext* globalCtx);
+
+extern FlexSkeletonHeader D_06000EE8[];
```
Comment (engineer124) [inline src/overlays/actors/ovl_Bg_Haka_Tomb/z_bg_haka_tomb.c:19]:
```suggestion
extern CollisionHeader D_06000EE8[];
```
This should fix the extra warning that was introduced and pass Jenkins

File: src/overlays/actors/ovl_Bg_Haka_Tomb/z_bg_haka_tomb.c [hunk 2]
```diff
@@ -21,26 +30,100 @@ const ActorInit Bg_Haka_Tomb_InitVars = {
...
     (ActorFunc)BgHakaTomb_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Haka_Tomb_0x80BD6580/BgHakaTomb_Init.asm")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+Vec3f D_80BD68A4 = { 30.0f, 90.0f, 0.0f };
```
Comment (engineer124) [inline src/overlays/actors/ovl_Bg_Haka_Tomb/z_bg_haka_tomb.c:38]:
```suggestion
static Vec3f D_80BD68A4 = { 30.0f, 90.0f, 0.0f };
```

File: src/overlays/actors/ovl_Bg_Haka_Tomb/z_bg_haka_tomb.c [hunk 3]
```diff
@@ -21,26 +30,100 @@ const ActorInit Bg_Haka_Tomb_InitVars = {
...
+    if (*arg0 >= 0 && arg2 > 0) {
+        s32 i = 0;
+        do {
+            if (*arg0 == arg1[i]) {
+                retVal = true;
+                break;
+            }
+            i++;
+
+        } while (i != arg2);
```
Comment (engineer124) [inline src/overlays/actors/ovl_Bg_Haka_Tomb/z_bg_haka_tomb.c:74]:
```suggestion
        s32 i;
        for (i = 0; i != arg2; i++) {
            if (*arg0 == arg1[i]) {
                retVal = true;
                break;
            }
        }
```
Now that this function matches nicely, would this match? Even better, would it match if you put the `s32 i` at the start? Even better, would this still match if you use `for (i = 0; i < arg2; i++) {`

**Review by engineer124 — COMMENTED** [review]
Nice work! Here are my comments

File: src/overlays/actors/ovl_Bg_Haka_Tomb/z_bg_haka_tomb.c (hunk 3 above)
Comment (kyleburnette) [reply src/overlays/actors/ovl_Bg_Haka_Tomb/z_bg_haka_tomb.c:74]: Oh wow, I was able to spruce this function up a bunch now you're right

File: src/overlays/actors/ovl_Bg_Haka_Tomb/z_bg_haka_tomb.c [hunk 4]
```diff
@@ -9,7 +9,16 @@ void BgHakaTomb_Destroy(Actor* thisx, GlobalContext* globalCtx);
...
 
-/*
+void func_80BD6624(BgHakaTomb* this);
+void func_80BD66AC(BgHakaTomb* this, GlobalContext* globalCtx);
+void func_80BD6754(BgHakaTomb* this);
+void func_80BD6768(BgHakaTomb* this, GlobalContext* globalCtx);
+void BgHakaTomb_SetupDoNothing(BgHakaTomb* this);
+void BgHakaTomb_DoNothing(BgHakaTomb* this, GlobalContext* globalCtx);
+
+extern CollisionHeader D_06000EE8[];
```
Comment (engineer124) [inline src/overlays/actors/ovl_Bg_Haka_Tomb/z_bg_haka_tomb.c:19]:
```suggestion
extern CollisionHeader D_06000EE8;
```
No brackets

**Review by engineer124 — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**